cellular_configs/v1alpha1/commoncoreranconfig_types.go

```diff
@@ -0,0 +1,103 @@
+)
+
+// CommonCoreRANStatus defines the observed state of CommonCoreRAN
+type CommonCoreRANConfigStatus struct {
+}
+
+// +kubebuilder:object:root=true
+type CommonCoreRANConfig struct {
+	metav1.TypeMeta   `json:",inline" yaml:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	Spec   CommonCoreRANConfigSpec   `json:"spec,omitempty" yaml:"spec,omitempty"`
+	Status CommonCoreRANConfigStatus `json:"status,omitempty" yaml:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// CommonCoreRANConfigList contains a list of CommonCoreRANConfigs
+type CommonCoreRANConfigList struct {
+	metav1.TypeMeta `json:",inline" yaml:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+	Items           []CommonCoreRANConfig `json:"items" yaml:"items"`
+}
+
+// CommonCoreRANConfigSpec defines the characteristics of a deployment of a network function
+type CommonCoreRANConfigSpec struct {
+	// plmnInfo defines the list of PLMNs to configure for core NFs
+	PlmnInfo []PlmnInfoConfig `json:"plmnInfo,omitempty" yaml:"plmnInfo,omitempty"`
+	// communicationModel defines the communcation strategy between core components, using NRF or without NRF
+	// +optional
+	CommunicationModel []CommunicationModelConfig `json:"communicationModel,omitempty" yaml:"communicationModel,omitempty"`
+}
+
+// PlmnInfoConfig defines the structure of PLMN
+type PlmnInfoConfig struct {
+	// mcc defines the mobile country code
+	// +kubebuilder:validation:Pattern=`[02-79][0-9][0-9]`
+	Mcc string `json:"mcc" yaml:"mcc"`
+	// mnc defines the mobile network code
+	// +kubebuilder:validation:Pattern=`[0-9][0-9][0-9]|[0-9][0-9]`
+	Mnc string `json:"mnc" yaml:"mnc"`
+	// tac defines the tracking area code
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=16777215
+	Tac int `json:"tac" yaml:"tac"`
+	// NSSAI defines the Network Slice Selection Assistance Information
+	// +optional
+	Nssai []Nssai `json:"nssai,omitempty" yaml:"nssai,omitempty"`
+}
+
+// NSSAI defines the Network Slice Selection Assistance Information
+type Nssai struct {
+	// Sst defines Slice/Service Type
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=255
+	Sst int `json:"sst" yaml:"sst"`
+	// Sd defines Service Differentiator
+	// +optional
+	// +kubebuilder:validation:Pattern=`^[A-Fa-f0-9]{6}$`
+	Sd string `json:"sd,omitempty" yaml:"sd,omitempty"`
+	// dnnInfo defines the Data Network Names information
+	DnnInfo []DnnInfo `json:"dnnInfo,omitempty" yaml:"dnnInfo,omitempty"`
+}
+
+// DNN defines the Data Network Names Information
+type DnnInfo struct {
+	// name defines dnn
+	Name string `json:"name" yaml:"name"`
+	// SessionType defines session type
+	// +kubebuilder:validation:Enum=ipv4;ipv6;ipv4v6;unstructured;ethernet
+	SessionType string `json:"sessionType" yaml:"sessionType"`
+	// Dns defines Domain Network Service Name
+	// +optional
+	Dns string `json:"dns" yaml:"dns"`
+}
+
```

Comment:
We need to associate a data network with a UE IP pool --- I don't see where we would get that information from this definition...

Reply:
@s3wong The UE-pool information for R2 we need to get it from `DNN` CR because there is no specializer for now that will fetch the information from DNN CR and mention it here. 

@henderiw this is what was proposed right? Only for R2?

Reply:
yes for R2 the name of the inn here should match the one in the nf deployment. It is clunky but it avoids doing all the other work required. So for R2 we use this hack for now.